ansible-playbooks/ros_install_and_set_up.yml

```diff
@@ -28,17 +28,41 @@
     - fail:
         msg: insights-client core collect is disabled.
       when: core_collect.msg == 'line added'
+    - debug: var=ansible_distribution
+    - debug: var=ansible_distribution_major_version
     - debug: var=ansible_architecture
-    - name: Enable required repository for x86_64 arch
+    - name: Enable required repository for RHEL8 x86_64 arch
       rhsm_repository:
         name: rhel-8-for-x86_64-baseos-rpms,rhel-8-for-x86_64-appstream-rpms
         state: enabled
-      when: ansible_architecture == "x86_64"
-    - name: Enable required repository for ARM arch
+      when:
+        - ansible_architecture == "x86_64"
+        - ansible_distribution == "RedHat"
+        - ansible_distribution_major_version == "8"
+    - name: Enable required repository for RHEL8 ARM arch
       rhsm_repository:
         name: rhel-8-for-aarch64-baseos-rpms,rhel-8-for-aarch64-appstream-rpms
         state: enabled
-      when: ansible_architecture == "aarch64"
+      when:
+        - ansible_architecture == "aarch64"
+        - ansible_distribution == "RedHat"
+        - ansible_distribution_major_version == "8"
+    - name: Enable required repository for RHEL7 x86_64 arch
+      rhsm_repository:
+        name: rhel-7-server-rpms
+        state: enabled
+      when:
+        - ansible_architecture == "x86_64"
+        - ansible_distribution == "RedHat"
+        - ansible_distribution_major_version == "7"
+    - name: Enable required repository for RHEL7 ARM arch
+      rhsm_repository:
+        name: rhel-7-server-rpms
+        state: enabled
+      when:
+        - ansible_architecture == "aarch64"
+        - ansible_distribution == "RedHat"
+        - ansible_distribution_major_version == "7"
```

Comment:
Here, using the same repository i.e rhel-7-server-rpms.
In case no additional changes required based on architecture, will it be good to have single condition based on distribution & major version?

Reply:
done.